src/events/processSignatureRequests.js

```diff
@@ -1,11 +1,36 @@
 require('dotenv').config()
 const Web3 = require('web3')
 const HttpListProvider = require('http-list-provider')
+const bridgeValidatorsABI = require('../../abis/BridgeValidators.abi')
 const logger = require('../services/logger')
 const rpcUrlsManager = require('../services/getRpcUrlsManager')
 const { createMessage } = require('../utils/message')
+const { AlreadyProcessedError, InvalidValidatorError } = require('../utils/errors')
 
 const { VALIDATOR_ADDRESS, VALIDATOR_ADDRESS_PRIVATE_KEY } = process.env
+const { HttpListProviderError } = HttpListProvider
+
+async function estimateGas(web3, bridgeContract, address, method, options) {
+  try {
+    const gasEstimate = await method.estimateGas(options)
+    return gasEstimate
+  } catch (e) {
+    if (e instanceof HttpListProviderError) {
+      throw e
+    }
+
+    const validatorContractAddress = await bridgeContract.methods.validatorContract().call()
+    const validatorContract = new web3.eth.Contract(bridgeValidatorsABI, validatorContractAddress)
```

Comment:
I think it would be better to get the validator address and contract instance only once instead of every time a transaction gas estimate fails

Reply:
Besides previous comment, the approach looks good to me